src/main/java/com/abdullah/coding/challenge/controllers/AdminController.java

```diff
@@ -0,0 +1,43 @@
+package com.abdullah.coding.challenge.controllers;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.CabDTO;
+import com.abdullah.coding.challenge.services.AdminService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("admin")
+public class AdminController {
+    @Autowired
+    private AdminService service;
+
+    @PatchMapping("management/cancel-booking")
+    public ResponseEntity<?> cancelBooking(@RequestParam("bookCode") String bookingCode) {
+        Response<?> response = service.cancelBooking(bookingCode);
+        return new ResponseEntity<>(response, response.getStatusCode());
+    }
+
+    @PatchMapping("management/assign-cab")
+    public ResponseEntity<?> assignCabToBooking(@RequestParam("id") int bookingId) {
+        Response<?> response = service.assignCab(bookingId);
+        return new ResponseEntity<>(response, response.getStatusCode());
+    }
+
+    @PatchMapping("management/cab-status")
+    public ResponseEntity<?> blockCab(@RequestParam("cabId") int cabId, @RequestParam("block") boolean blockStatus) {
+        CabDTO cabDTO = new CabDTO(cabId, blockStatus);
```

Comment:
**Why haven't we asked the client to send this DTO instead of the params in the request and create it by ourselves?**
- If by DTO you mean that if client should send the required info in a request body, It is possible but is it really required? Same thing can be achieved by using query. I did not see any reason to use request BODY as opposed to request params.